### ISSUE TITLE
SSO Login does not support FIDO keys

### ISSUE THREAD

**Issue by KramNamez** [RocketChat/Rocket.Chat#29784]
### Description:

To log in to the Rocket.Chat desktop app, I have to log in via our Keycloak instance for SSO.
Since I set up my FIDO keys (a NitroKey and a Yubikey) with PINs, I can no longer use them in this process. I have to switch over to the password-based login flow, where I can use the FIDO authenticators as 2FA - although the prompt to interact with them is still missing. Since the authenticator blinks for interaction, I still see the request and can do the user presence check.

Rocket.Chat should instead prompt me for the PIN for the FIDO keys and continue with the passwordless login flow (or at the very least prompt me when using them for 2FA).

### Steps to reproduce:

1. Set up a webauthn key for passwordless login in Keycloak and configure a PIN for it.
2. Try to use it to log in to Rocket.Chat.

### Expected behavior:

Rocket.Chat redirects to Keycloak, where I am prompted for the PIN and log in with my FIDO key.

### Actual behavior:

Rocket.Chat redirects to Keycloak, where I am not prompted for a PIN and my FIDO authenticators fail to allow for passwordless login. Fallback to 2FA without PIN also works improperly and does not prompt for interaction when required.

### Server Setup Information:

- Version of Rocket.Chat Server: 7.12.2
- Number of Running Instances: 1
- NodeJS Version: v22.16.0
- MongoDB Version: 8.0.13

### Client Setup Information

- Desktop App or Browser Version: Desktop
- Operating System: Pop!_OS

**Comment by KramNamez** [comment]
It turns out that with setting "attestation conveyance preference" to "none" and "user verification" to "discouraged", I can at least use the FIDO authenticators as 2FA in the desktop app. 

This is still a rather strange limitation, especially as the usual "touch your security key" popup from the browser fails to appear. You just have to know that it's working anyway.

Plus, it should also be able to ask for a PIN and allow for passwordless flows.

**Comment by EHEX-schildt** [comment]
We are also using a IdP with Protected(Pin) FIDO Key's and this is hard requirement for us.

If this don't get fixed, we have to move to another Product.

**Comment by KramNamez** [comment]
Last I asked, I was told that other features (like the accessibility enhancements) were higher priority. At least some of those have landed since then, and yet this isn't even tracked as a long-term issue. As far as I can tell, you have a fix (that you implemented for Google), which just needs to be applied more generally.

Is there any way I can convince you to track this as a medium-term issue, at least?

**Comment by 736-c41-2c1-e464fc974** [comment]
I can second this; FIDO Key support for user authentication is a hard requirement for us too. Without this, the desktop app is not usable for us.

**Comment by KlavsKlavsen** [comment]
I am evaluating rocket.chat - and SSO via Keycloak is also a requirement for us so i was reading a bit up on this :)

Keycloak uses the oauth2 protocol - so rocket.chat should never see how keycloak does the auth on its end (password + yubikey/totp - or whatever) - as it merely gets a redirect back with keycloak JWT? 

Is this perhaps because the desktop client is not opening up an "the auth url" - in 'the default browser" (which it should IMHO) - but doing it in some embedded browser instead which doesn't "work well" ?

**Comment by KramNamez** [comment]
Yup, just like the mobile app uses a WebView, this uses the Electron built-in - though weirdly enough, that _should_ support it. And as documented, it's not like WebauthN fundamentally is missing from that browser, it just fails to do _some parts_ of the necessary interactions. 

So the problem is that you cannot complete the login flow in Keycloak properly and never get redirected back to RocketChat. You _can_ login using Username+Password with a FIDO authenticator providing 2FA without a PIN, but even then, it's not working properly. It fails to tell you when to interact with the authenticator for user presence, but it at least does complete the flow.
